sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala

```diff
@@ -251,6 +251,11 @@ case class InsertIntoHiveTable(
       }
     }
 
+    //remove temporary directories
+    val fs = outputPath.getFileSystem(jobConf)
+    if ( outputPath.getParent.isRoot == false ) 
```

Comment:
I think the code uses `!` instead of `== false`. There shouldn't be space around the arg, and add the braces here too.

Why delete the root though? `outputPath` ought to be what you delete recursively.

I think you're correct given the comments though I don't know enough about this API to be sure.

Can you add `[SQL]` to the title please? normal format is `[SPARK-7300] [SQL] ...`